File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int n = blocks.size();
        int ans = INT_MAX;
        // no of operations -> no of swap to convert white into black
        // first window
        int curr_window = 0;
    
        for(int i=0; i<k; i++){
            if (blocks[i] == 'W'){
                curr_window++;
            }
        }
        ans = min(ans,curr_window);
        for(int j=k; j<n; j++){
            if(blocks[j] == 'W'){
                curr_window++;

            }
            if(blocks[j-k] == 'W'){
                curr_window--;
            }

            ans = min(ans,curr_window);
            
        }

        return ans;
    }
};

File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        n = len(blocks)
        ans = float('inf')
        curr_window = 0
        # first window
        for i in range(k):
            if blocks[i] == 'W':
                curr_window += 1
        ans = curr_window

        # slide the window
        for j in range(k,n):
            if blocks[j] == 'W':
                curr_window += 1
            if blocks[j-k] == 'W':
                curr_window -= 1
            ans = min(ans,curr_window)
        return ans
            